### DISCUSSION TITLE
Handling Cloudflare R2 Storage URLs with Self-Hosted Supabase

### DISCUSSION THREAD

**Discussion by abanobboles** [supabase/supabase#22534] [Questions]
Hello everyone,

I've set up a self-hosted Supabase instance and am using Cloudflare R2 as the storage solution (replacing S3). When I upload files through Supabase's storage API, it seems the files are first uploaded to my server and then forwarded to Cloudflare R2. However, the URLs I retrieve for these resources from Supabase point to my server's domain, not directly to Cloudflare R2.

This leads me to believe that requests for these files might be routed through my server, potentially consuming unnecessary bandwidth, instead of allowing direct access to the files stored on R2. My main question is whether this is indeed the case, and if so, how I can configure Supabase or Cloudflare R2 to provide direct file access URLs, bypassing my server after the initial upload.

Is there a way to configure the system so that the resource URLs provided by Supabase point directly to Cloudflare R2, or to otherwise ensure that file access does not unnecessarily consume my server's bandwidth?

Any insights or guidance on this would be greatly appreciated. Thank you in advance for your help!

**Comment by pickedbrian** [comment]
Hi, did you manage to solve the issue? @abanobboles
